lib/ash/resource/validation/compare.ex

```diff
@@ -225,6 +225,7 @@ defmodule Ash.Resource.Validation.Compare do
         |> Enum.map(fn {k, v} ->
           case v do
             fun when is_function(fun, 0) -> {k, fun.()}
+            %Decimal{} = v -> {k, v}
             v when is_list(v) -> {k, format_value(v)}
             v when is_map(v) -> {k, format_value(v)}
```

Comment:
The actual fix there should be `is_struct` to cover any struct since it's the `is_map` that's causing the bug.